library/cartridge_check_instance_started.py

```diff
@@ -4,32 +4,97 @@
 from ansible.module_utils.helpers import ModuleRes, CartridgeException
 from ansible.module_utils.helpers import get_control_console
 
-
 argument_spec = {
     'control_sock': {'required': True, 'type': 'str'},
     'stateboard': {'required': True, 'type': 'bool'},
+    'expected_states': {'required': False, 'type': 'list', 'default': ['Unconfigured', 'RolesConfigured']},
+    'check_buckets_are_discovered': {'required': False, 'type': 'bool', 'default': False},
 }
 
 
-def check_instance_started(params):
-    try:
-        control_console = get_control_console(params['control_sock'])
-        instance_is_alive = True
-        if not params['stateboard']:
-            instance_is_alive, _ = control_console.eval_res_err('''
-                return require('membership').myself().status == 'alive'
+def check_stateboard_started(control_console):
+    box_status, err = control_console.eval_res_err('''
+        if type(box.cfg) == 'function' or box.cfg.listen == nil then
+            return nil, "box hasn't been configured"
+        end
+        return true
+    ''')
+    if err is not None:
+        return ModuleRes(success=False, msg="Stateboard is not running: %s" % err)
+
+    return ModuleRes(success=True)
+
+
+def check_instance_started(control_console, expected_states, check_buckets_are_discovered):
+    instance_state, err = control_console.eval_res_err('''
+        return require('cartridge.confapplier').get_state()
+    ''')
+    if err is not None:
+        return ModuleRes(success=False, msg="Impossible to get state: %s" % err)
```

Comment:
`get_state()` can `return "OperationError, err" too. It wouldn't correspond to the message "Impossible to get state".